GVRf/Framework/framework/src/main/java/org/gearvrf/GVRScene.java

```diff
@@ -145,6 +145,7 @@ public void removeAllSceneObjects() {
 
         rig.removeAllChildren();
         for (GVRSceneObject child : mSceneRoot.getChildren()) {
+            child.detachAllComponents();
```

Comment:
Why did you need to add this line?

I believe because of it the tests started crashing left and right. It is a framework problem and not an issue with your change. Currently nothing can be done other than sprinkling nullptr checks everywhere.

Reply:
@ragner I am about to remove the detachAllComponents from removeAllSceneObjects. Let me know if you have concerns. If you are to reuse scene objects after removeAllSceneObject detachAllComponents would break the reused scene objects.

Reply:
I believe audio will continue playing after removing scene objects. We can fix that after you finish this change.

Reply:
Unfortunately there is no good general and lightweight mechanism to know when a scene object starts and stops being rendered. While something could be added to removeSceneObject on the Java side to let scene objects know they are no longer parented to the main scene root it won't be very lightweight. Nor complete - since addSceneObject then should be producing similar notification. And then the main scene could change too. On the renderer side there is a bit more flexibility but it has its own complexity. And making a ton of JNI callbacks to Java might not be the best idea from the render thread. Nola may have better ideas.

I think for now GVRAudioSource's javadoc needs to mention that it is up to the user to stop the audio playback whenever appropriate.

Reply:
There is a notification on the C++ side. Component::onAddedToScene and Component::onRemovedFromScene are called. I did not know how to notify Java. I think we need a way to communicate events from C++ to Java asynchronously.